hydra-node/test/Hydra/ModelSpec.hs

```diff
@@ -176,6 +176,34 @@ spec = do
   prop "implementation respects model" $ forAll arbitrary prop_checkModel
   prop "check conflict-free liveness" prop_checkConflictFreeLiveness
   prop "check head opens if all participants commit" prop_checkHeadOpensIfAllPartiesCommit
+  prop "fanout contains whole confirmed UTxO" prop_fanoutContainsWholeConfirmedUTxO
+
+prop_fanoutContainsWholeConfirmedUTxO :: Property
+prop_fanoutContainsWholeConfirmedUTxO =
+  forAllDL fanoutContainsWholeConfirmedUTxO prop_HydraModel
+
+-- | Given any random walk of the model, if the Head is open a NewTx getting
+-- confirmed must be part of the UTxO after finalization.
+fanoutContainsWholeConfirmedUTxO :: DL WorldState ()
+fanoutContainsWholeConfirmedUTxO = do
+  anyActions_
+  getModelStateDL >>= \case
+    st@WorldState{hydraState = Open{}} -> do
+      (party, payment) <- forAllNonVariableQ (nonConflictingTx st)
+      tx <- action $ Model.NewTx party payment
+      eventually (ObserveConfirmedTx tx)
+      action_ $ Model.Close party
+      void $ action $ Model.Fanout party
```

Comment:
It is a bit hard to do that since what we get here is wrapped in a `Var` and also the `tx` is `Payment` so it needs to be converted into `UTxO` and then somehow compared or find another workaround.

Reply:
This is what we would have wanted, but Quvic people said that this is currently not possible. The alternative would be an assertive action `AssertFanoutUTxO` or so, which then would do the check in the `postcondition` (or `perform`). We ended up doing this in all `Fanout` actions.

A comment should be added here though that the check is actually happening in the postcodition.